src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -451,13 +402,378 @@ def __init__(
+        self._electrical_series_name = electrical_series_name
+
+        self.file_path = file_path
+        hdf5_file = _read_hdf5_file(
+            file_path=file_path,
+            file=file,
+            stream_mode=stream_mode,
+            cache=cache,
+            stream_cache_path=stream_cache_path,
+        )
+
+        # If the electrical_series_location is not given, `find_electrical_series` will be called
+        # And returns a dictionary with the electrical_series_name as key and the location as value
+        # If there is only one electrical series, the electrical_series_name is set to the name of the series
+        if electrical_series_location is None:
+            available_electrical_series = _NWBHDF5RecordingExtractor.find_electrical_series(hdf5_file)
+            if electrical_series_name is None:
+                if len(available_electrical_series) == 1:
+                    electrical_series_name = list(available_electrical_series.keys())[0]
+                else:
+                    raise ValueError(
+                        "Multiple ElectricalSeries found in the file. "
+                        "Please specify the 'electrical_series_name' argument:"
+                        f"Available options are: {available_electrical_series}."
+                    )
+            else:
+                if electrical_series_name not in available_electrical_series:
+                    raise ValueError(
+                        f"'{electrical_series_name}' not found in the file. "
+                        f"Available options are: {available_electrical_series}"
+                    )
+            electrical_series_location = available_electrical_series[electrical_series_name]
+
+        self.electrical_series_location = electrical_series_location
+
+        # The indices in the electrode table corresponding to this electrical series
+        electrical_series = hdf5_file[electrical_series_location]
+        electrodes_indices = electrical_series["electrodes"][:]
+        # The table for all the electrodes in the nwbfile
+        electrodes_table = hdf5_file["/general/extracellular_ephys/electrodes"]
+        electrode_table_columns = electrodes_table.attrs["colnames"]
+
+        # Get sampling frequency
+        if "starting_time" in electrical_series.keys():
+            t_start = electrical_series["starting_time"][()]
+            sampling_frequency = electrical_series["starting_time"].attrs["rate"]
+        elif "timestamps" in electrical_series.keys():
+            t_start = electrical_series["timestamps"][0]
+            sampling_frequency = 1 / np.median(np.diff(electrical_series["timestamps"][:1000]))
+
+        # TimeSeries need to have either timestamps or rate
+        if sampling_frequency is None:
+            raise ValueError(
+                "Could not find rate information as both 'rate' and "
+                "'timestamps' are missing from the file. "
+                "Use the 'sampling_frequency' argument."
```

Comment:
I don't think the `sampling_frequency` is among the available args (and it should not be).

In this case, we should just say that the electrical series is not valid

Reply:
Same logic as https://github.com/SpikeInterface/spikeinterface/pull/2298/files#diff-94bcb22069dd20ece6a264cf92514e386c25f0851afa2af80cbc7cc5bcba339bR254-R260

Reply:
I am fine with fixing this but note that I just copied from main:
https://github.com/h-mayorquin/spikeinterface/blob/3702e9fbc11e8317c815cbdfa44662e22d01f5e2/src/spikeinterface/extractors/nwbextractors.py#L300-L307

Reply:
Ah I think that was copied form the Sorting which has the `sampling_frequency` argument! Let's change both messages to:
```
Invalid ElectricalSeries! Could not find rate information as both 'rate' and''timestamps' are missing from the file.
```

Reply:
I am just eliminate the assertion, it can't happen. Either it has timestamps or rate.